os/services/shell/shell.c

```diff
@@ -130,6 +130,9 @@ PT_THREAD(shell_input(struct pt *pt, shell_output_func output, const char *cmd))
     args++;
   }
 
+  /* ignore empty lines */
+  if(strlen(cmd)==0) goto done;
```

Comment:
Code style: even single-line blocks must be within `{}`
Goto: we generally try to have them only if it simplifies the flow significantly. In this case I'd personally find it neater to just put the command lookup in an `if` block.

Reply:
Got rid of both uses of goto.